### PR TITLE
Docs: Add .env instruction for critical secure

### PR DESCRIPTION
**Pull request by miguelgargallo**
Docs: Add .env instruction for critical secure

### CODE REVIEW

File: app/templates/docs/integration/sdks/nodejs.hbs [hunk 1]
```diff
@@ -41,6 +41,69 @@
 
 <Layout::Gap mini />
 
+<Layout::Text text-l text-f>
+	<h3>Extra Configuration .env file</h3>
+	<p>First install dotenv:</p>
```
Comment (Ekwuno) [inline app/templates/docs/integration/sdks/nodejs.hbs:46]:
```suggestion
	<p> To do so, first install <Link @link="https://www.npmjs.com/package/dotenv">dotenv</Link>  which is an NPM package that loads environment variables from a  <code> .env </code> file</p>
```

File: app/templates/docs/integration/sdks/nodejs.hbs [hunk 2]
```diff
@@ -41,6 +41,69 @@
...
+	<Code @name="docs-integration-libraries-nodejs-extra-install-dotenv.bash">
+		npm install dotenv
+	</Code>
+	<p>Add a<code>.env</code>file:</p>
```
Comment (Ekwuno) [inline app/templates/docs/integration/sdks/nodejs.hbs:50]:
```suggestion
	<p>Add a <code>.env</code> file to the root of your project. Then input your SurrealDB credentials</p>
```

File: app/templates/docs/integration/sdks/nodejs.hbs [hunk 3]
```diff
@@ -41,6 +41,69 @@
...
+	<Code @name="docs-integration-libraries-nodejs-extra-config.bash">
+		SURREALDB_LINK=https://cloud.surrealdb.com/rpc
+		SURREALDB_USERNAME=root
+		SURREALDB_PASSWORD=root
+		SURREALDB_NAMESPACE=surrealdb
+		SURREALDB_DATABASE=docs
+	</Code>
+	<p>Then you use it as follows:</p>
```
Comment (Ekwuno) [inline app/templates/docs/integration/sdks/nodejs.hbs:58]:
```suggestion
	<p>Then in your `app.js` file  edit the values as follows:</p>
```

File: app/templates/docs/integration/sdks/nodejs.hbs [hunk 4]
```diff
@@ -41,6 +41,69 @@
...
 
+<Layout::Text text-l text-f>
+	<h3>Extra Configuration .env file</h3>
+	<p>First install dotenv:</p>
+	<Code @name="docs-integration-libraries-nodejs-extra-install-dotenv.bash">
+		npm install dotenv
+	</Code>
+	<p>Add a<code>.env</code>file:</p>
+	<Code @name="docs-integration-libraries-nodejs-extra-config.bash">
+		SURREALDB_LINK=https://cloud.surrealdb.com/rpc
```
Comment (Ekwuno) [inline app/templates/docs/integration/sdks/nodejs.hbs:52]:
```suggestion
		SURREALDB_LINK=http://127.0.0.1:8000/rpc
```

File: app/templates/docs/integration/sdks/nodejs.hbs [hunk 5]
```diff
@@ -41,6 +41,69 @@
...
+		const { default: Surreal } = require('surrealdb.js');
+		require('dotenv').config();
+
+		let surrealdblink = process.env.SURREALDB_LINK;
+		let username = process.env.SURREALDB_USERNAME;
+		let password = process.env.SURREALDB_PASSWORD;
+		let namespace = process.env.SURREALDB_NAMESPACE;
+		let database = process.env.SURREALDB_DATABASE;
+
+		const db = new Surreal(surrealdblink); // Connect to SurrealDB but on your .env file
```
Comment (Ekwuno) [inline app/templates/docs/integration/sdks/nodejs.hbs:69]:
```suggestion
		const db = new Surreal(surrealdburl); // Connect to SurrealDB 
```

**Review by Ekwuno — CHANGES_REQUESTED** [review]
Thanks for sending in this PR. I have just done a first review.

**Comment by Ekwuno** [comment]
@miguelgargallo will you add the suggestions in this PR? I see you have marked them as resolved but it's not been added to the PR.

**Comment by Ekwuno** [comment]
The same changes apply to #203 as well.

**Comment by miguelgargallo** [comment]
changes will be applied in #203 in a while! 💜

**Comment by miguelgargallo** [comment]
> @miguelgargallo will you add the suggestions to another commit for this PR? I see you have marked them as resolved but it's not been added to the PR.

applying to this right now, my apologies

**Comment by miguelgargallo** [comment]
- [x] done

File: app/templates/docs/integration/sdks/nodejs.hbs [hunk 6]
```diff
@@ -41,6 +41,71 @@
...
+	<h3>Additional Configuration for credentials - Optional </h3>
+	
+	When handling credentials in applications,  It is recommended that you add these to a `.env` file 
+	<p> To do so, first install <Link @link="https://www.npmjs.com/package/dotenv">dotenv</Link>  which is an NPM package that loads environment variables from a  <code> .env </code> file</p>
+	<Code @name="docs-integration-libraries-nodejs-extra-install-dotenv.bash">
+		npm install dotenv
+	</Code>
+	<p>Add a <code>.env</code> file to the root of your project. Then input your SurrealDB credentials</p>
+	<Code @name="docs-integration-libraries-nodejs-extra-config.bash">
+		SURREALDB_LINK=http://127.0.0.1:8000/rpc
```
Comment (Ekwuno) [inline app/templates/docs/integration/sdks/nodejs.hbs:54]:
```suggestion
		SURREALDB_URL=http://127.0.0.1:8000/rpc
```

**Comment by kearfy** [comment]
Hi @miguelgargallo, copying over my response from #203 here!

I have taken a look at your suggested change but decided that we will not be merging this. While it absolutely is good practice to maintain connection details and similar in environment variables, it is an opinionated solution. We want to keep the documentation for SurrealDB and it's SDKs as abstract as possible, and don't feel like this is in line with that goal.

In any case, thanks for your contribution! It is much appreciated!

**Comment by miguelgargallo** [comment]
It's nothing you are welcome, see you at SurrealDB World 🥰🚀👩‍🚀🌠

**Pull request closed without merging**